AIO/src/org/aio/activities/quests/TheRestlessGhost.java

```diff
@@ -0,0 +1,190 @@
+    private static final Area COFFIN = new Area(3247, 3195, 3252, 3190);
+    private static final Area WIZARD = new Area(3116, 9564, 3121, 9569);
+
+    private static final String[] AERECT_OPTIONS = {
+            "I'm looking for a quest!",
+            "Ok, let me help then."
+    };
+
+    private static final String[] URHNEY_OPTIONS = {
+            "Father Aereck sent me to talk to you.",
+            "He's got a ghost haunting his graveyard."
+    };
+
+    private static final String[] GHOST_OPTIONS = {
+            "Yep, now tell me what the problem is."
+    };
+
+    private static final int INVENTORY_SLOTS_REQUIRED = 4;
+
+    private static final String[] ITEMS_NEEDED = {
+            "Ghost's skull",
+            "Ghostspeak amulet",
+    };
+
+    private final DepositAllBanking depositAllBanking = new DepositAllBanking(ITEMS_NEEDED);
+
+    public TheRestlessGhost() {
+        super(Quest.THE_RESTLESS_GHOST);
+    }
+
+    @Override
+    public void onStart() {
+        depositAllBanking.exchangeContext(getBot());
+    }
+
+    @Override
+    public void runActivity() throws InterruptedException {
+        if (!getInventory().contains(ITEMS_NEEDED) && getInventory().getEmptySlotCount() < INVENTORY_SLOTS_REQUIRED) {
+            depositAllBanking.run();
+        } else if (getTabs().getOpen() != Tab.INVENTORY) {
+            getTabs().open(Tab.INVENTORY);
+        } else {
+            switch (getProgress()) {
+                case 0:
+                    talkToAereck();
+                    break;
+                case 1:
+                    talkToUrhney();
+                    break;
+                case 2:
+                    talkToGhost();
+                    break;
+                case 3:
+                    searchAlter();
+                    break;
+                case 4:
+                    placeSkull();
+                    break;
+                case 5:
+                    // Make sure we are not in the cut scene
+                    if (getWidgets().get(277, 2).isVisible()) {
+                        log("Quest is complete");
+                        isComplete = true;
+                    }
+                    break;
+                default:
+                    log("Unknown progress config value: " + getProgress());
+                    setFailed();
+                    break;
+            }
+        }
+    }
+
+    private void placeSkull() throws InterruptedException {
+        RS2Object coffin = getObjects().closest("Coffin");
+
+        if (!COFFIN.contains(myPosition())) {
+            getWalking().webWalk(COFFIN);
+        } else if(coffin != null && coffin.hasAction("Close") && getInventory().contains("Ghost's skull")) {
+            useSkull();
+        } else if (coffin != null && coffin.hasAction("Open")) {
+            openCoffin();
+        }
+    }
+
+    private void useSkull() throws InterruptedException {
+        if(getInventory().isItemSelected()) {
```

Comment:
We should use:

`if(!"Ghost's skull".equals(getInventory().getSelectedItemName()))`

Reply:
Ahh, yes I didn't know of a way to check specific item.